misc/package/Data Files/MWSE/core/lib/logger/init.lua

```diff
@@ -0,0 +1,1016 @@
+	-- bright white
+	[LOG_LEVEL.TRACE] = "\x1B[0m\x1B[1m\x1B[37mT\x1B[0m",
+}
+
+-- Define various things that will be moved into the official documtation system
+-- once the API has been finalized.
+do 
+
+--- A function that handles the formatting of log messages, given the `logger`, a `record`, and the log function parameters.
+--- The record holds information about the logging level and line number, and the rest of the parameters are the ones passed
+--- as arguments to the `Logger:debug` methods.
+--- This function should return a string, which will then be printed to the appropriate file.
+--- This function should also include the header of the log message. The default header can be accessed by 
+--- using the `makeHeader` method.
+---@alias Logger.formatter fun(self: Logger, record: Logger.Record, ...: string|any|fun(...): ...): string
+
+
+--- A logging record. Contains information about the log level and the line number.
+---@class Logger.Record
+---@field stackLevel integer how far up the stack we're being called
+---@field level Logger.LEVEL logging level
+---@field lineNumber integer|false the line number, if enabled for this logger
+---@field timestamp number|false the timestamp of this message
+
+--- Parameters used to create a new `Logger`. 
+-- All parameters are optional, as they will be retrieved automatically if not provided.
+-- 
+--- In addition to displayed parameters, you can also pass a `modName`, a `filepath`, a `level` and a `filepath`.
+-- Although for the latter options, it's recommended you use the relevant methods instead.
+---@class Logger.newParams
+---@field logToConsole bool? Should the output be written to the console?
+---@field modName string? the name of the mod this logger is for. will be automatically retrieved if not provided
+---@field modDir string? The directory that this mod operates in. Used to retrieve the `Mod_Info`.
+---@field moduleName string|false? The name of a module that this logger belongs to.
+-- This is useful in cases where one file has several distinct parts. 
+-- The `moduleName` will be displayed next to the name of the mod, in parentheses.
+---@field filePath string? path to the file. will be retrieved if not provided
+---@field level Logger.LEVEL? the log level to set this object to. Default: "LEVEL.INFO"
+---@field writeToFile string|boolean|nil whether to write the log messages to a file, or the name of the file to write to. if `false` or `nil`, messages will be written to `MWSE.log`
+---@field includeLineNumber boolean? Should the current line number be printed when writing log messages? Default: `true`
+---@field includeTimestamp boolean? should timestamps be included in logging messages? Default: `false`
+---@field abbreviateHeader boolean? Should the header messages be abbreviated? Default: `false`.
+---@field formatter Logger.formatter? A way to specify how logging messages should be formatted. A default one will be used if not provided.
+
```

Comment:
This parameter list is missing `logToConsole`. Please mention the option when writing the documentation.